### PR TITLE
Docs: update annotations docs to include typeahead support for tags

### PR DESCRIPTION
**Pull request by axelavargas**
**What this PR does / why we need it**:

Updates documentation for annotations and new HTTP endpoint, now we support typeahead suggestions, images are in these PR https://github.com/grafana/website/pull/5196

**Which issue(s) this PR fixes**:

For #36107

**Special notes for your reviewer**:

### CODE REVIEW

File: docs/sources/http_api/annotations.md [hunk 1]
```diff
@@ -251,3 +251,37 @@ Content-Type: application/json
...
+    "result": {
+        "tags": [
+            {
+                "tag": "outage",
+                "count": 1
+            }
+        ]
+    }
+}
+```
```
Comment (marefr) [inline docs/sources/http_api/annotations.md:287]: A bit unsure if the tag-query is optional or not. Please verify.

```suggestion
Find all the event tags created in the annotations

**Example Request**:

```http
GET /api/annotations/tags?tag=out HTTP/1.1
Accept: application/json
Content-Type: application/json
Authorization: Basic YWRtaW46YWRtaW4=
```

Query Parameters:
- `tag`: string. Optional. Used to filter tags.
- `limit`: number. Optional - default is 100. Max limit for results returned.

**Example Response**:

```http
HTTP/1.1 200
Content-Type: application/json

{
    "result": {
        "tags": [
            {
                "tag": "outage",
                "count": 1
            }
        ]
    }
}
```
```

**Review by marefr — CHANGES_REQUESTED** [review]
Suggested some minor changes

File: docs/sources/http_api/annotations.md (hunk 1 above)
Comment (axelavargas) [reply docs/sources/http_api/annotations.md:287]: @marefr Thank you for taking a look 🙌🏾, indeed the tag query is optional (I tested it), I will apply the suggestions

**Review by marefr — APPROVED** [review]
From technical standpoint LGTM, but would ask for technical writers input as well

File: docs/sources/http_api/annotations.md [hunk 2]
```diff
@@ -251,3 +251,41 @@ Content-Type: application/json
...
+```http
+GET /api/annotations/tags?tag=out HTTP/1.1
+Accept: application/json
+Content-Type: application/json
+Authorization: Basic YWRtaW46YWRtaW4=
+```
+
+Query Parameters:
+
+- `tag`: string. Optional. Used to filter tags.
```
Comment (achatterjee-grafana) [inline docs/sources/http_api/annotations.md:272]:
```suggestion
- `tag`: Optional. A string that you can use to filter tags.
```

File: docs/sources/http_api/annotations.md [hunk 3]
```diff
@@ -251,3 +251,41 @@ Content-Type: application/json
...
+- `limit`: number. Optional - default is 100. Max limit for results returned.
```
Comment (achatterjee-grafana) [inline docs/sources/http_api/annotations.md:273]:
```suggestion
- `limit`: Optional. A number, where the default is 100. Max limit for results returned.
```

File: docs/sources/dashboards/annotations.md [hunk 4]
```diff
@@ -41,12 +41,16 @@ can still show them if you add a new **Annotation Query** and filter by tags. Bu
...
+
+You can create new annotation queries that fetch annotations from the native annotation store via the `-- Grafana --` data source and by setting *Filter by* to `Tags`. In Grafana v8.1+ it's possible to get typeahead support of existing tags, you should specify at least one tag.
+
+For example create an annotation query name `outages` and specify a tag named `outage`. This query will show all annotations you create (from any dashboard or via API) that have the `outage` tag. By default, if you add multiple tags in the annotation query, Grafana will only show annotations that have all the tags you supplied. You can invert the behavior by enabling `Match any` which means that Grafana will show annotations that contains at least one of the tags you supplied.
```
Comment (achatterjee-grafana) [inline docs/sources/dashboards/annotations.md:47]:
```suggestion
You can create new queries to fetch annotations from the native annotation store via the `-- Grafana --` data source by setting *Filter by* to `Tags`. 

Grafana v8.1 and later versions also support typeahead of existing tags provided you have defined at least one tag.

For example, create an annotation query name `outages` and specify a tag `outage`. This query will show all annotations (from any dashboard or via API) with the `outage` tag. If multiple tags are defined in an annotation query, then Grafana will only show annotations matching all the tags. To modify the behavior, enable `Match any`, and Grafana will show annotations that contain any one of the tags you provided.
```

**Review by achatterjee-grafana — COMMENTED** [review]
Added doc review.

File: docs/sources/http_api/annotations.md [hunk 5]
```diff
@@ -251,3 +251,41 @@ Content-Type: application/json
...
+**Example Request**:
+
+```http
+GET /api/annotations/tags?tag=out HTTP/1.1
+Accept: application/json
+Content-Type: application/json
+Authorization: Basic YWRtaW46YWRtaW4=
+```
+
+Query Parameters:
```
[the remark this answers is not in the document]
Comment (axelavargas) [reply docs/sources/http_api/annotations.md:270]: 🤔 I don't think so, the Query parameters section is only describing a list of parameters that the API accepts.

**Comment by axelavargas** [comment]
> Added doc review.

Thank you for the review @achatterjee-grafana, I have updated the docs with your suggestions and also updated the reference of the images :).

File: docs/sources/http_api/annotations.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (achatterjee-grafana) [reply docs/sources/http_api/annotations.md:270]: Fair enough.

**Review by achatterjee-grafana — APPROVED** [review]
LGTM

**Pull request merged**